### PR TITLE
RGRIDT 723 - Improve the public API with more actions II

### PR DESCRIPTION
**Pull request by JoaoFerreira-FrontEnd**
### What was happening
* Public methods related with styling were not implemented on the API

### What was done
* Created the _**SetCellCssClass**_ method;
* Created the _**SetColumnCssClass**_ method;
* Created the _**RemoveAllCssClassesFromCell**_ method;
* Created the _**RemoveColumnCssClass**_ method;

### Test Steps
1. Add an onChange client action on one column;
2. Use the **AddCellCssClass** API action to change other cell appearance;
3. Add another onChange client action on another column;
4. Use the **RemoveAllCssClassesFromCell** API action to change other cell appearance;
5. Interact with the grid and check if the class is added on the first moment;
6. Then check if it's removed;
#
1. Add an onChange client action on one column;
2. Use the **AddColumnCssClass** API action to change other collumn appearance;
3. Add another onChange client action on another column;
4. Use the **RemoveCssClassesFromColumn** API action to change other collumn appearance with **_ApplyToHeader_** parameter (false);
5. Interact with the grid and check if the class is added on the first moment but the header cell of the column is not affected;
6. Then check if it's removed;
#
1. Add an onChange client action on one column;
2. Use the **AddColumnCssClass** API action to change other collumn appearance with **_ApplyToHeader_** parameter (true);
3. Add another onChange client action on another column;
4. Use the **RemoveCssClassesFromColumn** API action to change other collumn appearance;
5. Interact with the grid and check if the class is added on the first moment including the header cell;
6. Then check if it's removed;

### Checklist
* [x] tested locally
* [x] documented the code
* [x] clean all warnings and errors of eslint
* [x] requires changes in OutSystems 
* [x] requires new sample page in OutSystems

### CODE REVIEW

File: code/src/GridAPI/ContextMenu.ts [hunk 1]
```diff
@@ -17,8 +17,9 @@ namespace GridAPI.ContextMenu {
     ): string {
         //Try to find in DOM only if not present on Map
         if (lookUpDOM && !_menuItemsToGridId.has(menuItemId)) {
-            const menuOptionElement =
-                OSFramework.Helper.GetElementByUniqueId(menuItemId);
+            const menuOptionElement = OSFramework.Helper.GetElementByUniqueId(
```
Comment (glundgren93) [inline code/src/GridAPI/ContextMenu.ts:20]: Please use prettier here.

File: code/src/GridAPI/Styling.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+     * @param {string} cssClass
+     * @param {boolean} applyToHeader
+     */
+    export function SetColumnCssClass(
+        gridID: string,
+        columnID: string,
+        cssClass: string,
+        applyToHeader: boolean
+    ): void {
+        if (applyToHeader) {
```
Comment (glundgren93) [inline code/src/GridAPI/Styling.ts:44]: What happens if grid is not found?

File: code/src/GridAPI/Styling.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            GridManager.GetGridById(gridID).features.styling.addCssClassAll(
+                columnID,
+                cssClass
+            );
+        } else {
+            GridManager.GetGridById(gridID).features.styling.addCssClass(
```
Comment (glundgren93) [inline code/src/GridAPI/Styling.ts:50]: What happens if grid is not found?

File: code/src/GridAPI/Styling.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+     * @param {string} columnID
+     * @param {number} rowIndex
+     */
+    export function RemoveAllCssClassesFromCell(
+        gridID: string,
+        columnID: string,
+        rowIndex: number
+    ): void {
+        const binding = ColumnManager.GetColumnById(columnID).config.binding;
+        GridManager.GetGridById(gridID).features.cellStyle.removeClass(
```
Comment (glundgren93) [inline code/src/GridAPI/Styling.ts:70]: What happens if grid is not found?

File: code/src/GridAPI/Styling.ts [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+     * @param {string} gridID
+     * @param {string} columnID
+     * @param {number} rowIndex
+     */
+    export function RemoveAllCssClassesFromCell(
+        gridID: string,
+        columnID: string,
+        rowIndex: number
+    ): void {
+        const binding = ColumnManager.GetColumnById(columnID).config.binding;
```
Comment (glundgren93) [inline code/src/GridAPI/Styling.ts:69]: What happens if column doesnt exist?

File: code/src/GridAPI/Styling.ts [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+     * @param {string} gridID
+     * @param {string} columnID
+     * @param {string} cssClass
+     */
+    export function RemoveColumnCssClass(
+        gridID: string,
+        columnID: string,
+        cssClass: string
+    ): void {
+        GridManager.GetGridById(gridID).features.styling.removeCssClassAll(
```
Comment (glundgren93) [inline code/src/GridAPI/Styling.ts:88]: What happens if grid is not found?

File: code/src/GridAPI/Styling.ts [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+            columnID,
+            cssClass
+        );
+        GridManager.GetGridById(gridID).features.styling.removeCssClass(
```
Comment (glundgren93) [inline code/src/GridAPI/Styling.ts:92]: What happens if grid is not found?

File: code/src/OSFramework/Feature/IStyling.ts [hunk 8]
```diff
@@ -1,6 +1,30 @@
...
+         * @param {string} columnID
+         * @param {string} className
+         */
+        removeCssClass(columnID: string, className: string): void;
+        /**
+         * Removes a specific CSS Class from an entire column
+         * @param {string} columnID
+         * @param {string} className
+         */
+        removeCssClassAll(columnID: string, className: string): void;
```
Comment (glundgren93) [inline code/src/OSFramework/Feature/IStyling.ts:28]: Can you rename this method so we know it removes css classes from a column?

File: code/src/OSFramework/Feature/IStyling.ts [hunk 9]
```diff
@@ -1,6 +1,30 @@
...
+         * @param {string} className
+         */
+        addCssClassAll(columnID: string, className: string): void;
         changeRowHeight(rowHeight: number): void;
+        /**
+         * Removes a specific CSS Class from a column content
+         * @param {string} columnID
+         * @param {string} className
+         */
+        removeCssClass(columnID: string, className: string): void;
```
Comment (glundgren93) [inline code/src/OSFramework/Feature/IStyling.ts:22]: Can you rename this method so we know it removes css classes from a column?

File: code/src/OSFramework/Feature/IStyling.ts [hunk 10]
```diff
@@ -1,6 +1,30 @@
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 namespace OSFramework.Feature {
     export interface IStyling {
+        /**
+         * Add a specific CSS Class to the content of a column (header).
+         * @param {string} columnID
+         * @param {string} className
+         */
+        addCssClass(columnID: string, className: string): void;
```
Comment (glundgren93) [inline code/src/OSFramework/Feature/IStyling.ts:9]: Can you rename this method so we know it add css classes to a column?

File: code/src/OSFramework/Feature/IStyling.ts [hunk 11]
```diff
@@ -1,6 +1,30 @@
...
+        /**
+         * Add a specific CSS Class to a column (header and content).
+         * @param {string} columnID
+         * @param {string} className
+         */
+        addCssClassAll(columnID: string, className: string): void;
```
Comment (glundgren93) [inline code/src/OSFramework/Feature/IStyling.ts:15]: Can you rename this method so we know it addcss classes to a column?

File: code/src/WijmoProvider/Features/Styling.ts [hunk 12]
```diff
@@ -12,10 +12,50 @@ namespace WijmoProvider.Feature {
...
+
+            return classList;
+        }
+
         public get rowHeight(): number {
             return this._rowHeight;
         }
 
+        public addCssClass(columnID: string, className: string): void {
+            const classList = this._getCssClass(columnID);
```
Comment (glundgren93) [inline code/src/WijmoProvider/Features/Styling.ts:38]: What happens if column does not exist?

File: code/src/WijmoProvider/Features/Styling.ts [hunk 13]
```diff
@@ -12,10 +12,50 @@ namespace WijmoProvider.Feature {
...
+            if (index === -1) {
+                classList.push(className);
+                this._grid.getColumn(
+                    columnID
+                ).provider.cssClass = classList.join(' ');
+            }
+        }
+
+        public addCssClassAll(columnID: string, className: string): void {
+            const classList = this._getCssClassAll(columnID);
```
Comment (glundgren93) [inline code/src/WijmoProvider/Features/Styling.ts:49]: What happens if column does not exist?

File: code/src/WijmoProvider/Features/Styling.ts [hunk 14]
```diff
@@ -32,5 +72,30 @@ namespace WijmoProvider.Feature {
...
         }
+
+        public removeCssClass(columnID: string, className: string): void {
+            const classList = this._getCssClass(columnID);
+
+            const index = classList.indexOf(className);
+            if (index > -1) {
+                classList.splice(index, 1);
+            }
+            this._grid.getColumn(columnID).provider.cssClass = classList.join(
```
Comment (glundgren93) [inline code/src/WijmoProvider/Features/Styling.ts:83]: What happens if column does not exist?

File: code/src/WijmoProvider/Features/Styling.ts [hunk 15]
```diff
@@ -32,5 +72,30 @@ namespace WijmoProvider.Feature {
...
+
+        public removeCssClassAll(columnID: string, className: string): void {
+            const classList = this._getCssClassAll(columnID);
+
+            const index = classList.indexOf(className);
+            if (index > -1) {
+                classList.splice(index, 1);
+            }
+
+            this._grid.getColumn(
```
Comment (glundgren93) [inline code/src/WijmoProvider/Features/Styling.ts:96]: What happens if column does not exist?

**Review by glundgren93 — CHANGES_REQUESTED** [review]

File: code/src/GridAPI/ContextMenu.ts (hunk 1 above)
Comment (bmartinho) [inline code/src/GridAPI/ContextMenu.ts:20]: Have you ran npm run lint?

File: code/src/OSFramework/Feature/IStyling.ts (hunk 11 above)
Comment (bmartinho) [inline code/src/OSFramework/Feature/IStyling.ts:15]: In the OSFramework layer, can we have only one action: addColumnCSSClass(columnID, className, includeHeader)
?

File: code/src/OSFramework/Feature/IStyling.ts (hunk 8 above)
Comment (bmartinho) [inline code/src/OSFramework/Feature/IStyling.ts:28]: In the OSFramework layer, can we have only one action: removeColumnCSSClass(columnID, className, includeHeader)

**Review by bmartinho — CHANGES_REQUESTED** [review]

**Review by glundgren93 — APPROVED** [review]
LGTM

File: code/src/WijmoProvider/Features/Styling.ts [hunk 16]
```diff
@@ -12,10 +12,59 @@ namespace WijmoProvider.Feature {
...
+            let classList = [];
+            const column = this._grid.getColumn(columnID);
+
+            if (column) {
+                const classListString = column.provider.cssClassAll || ' ';
+                classList = classListString.split(' ');
+            }
+
+            return classList;
+        }
```
Comment (tmbp95) [inline code/src/WijmoProvider/Features/Styling.ts:37]: I believe these 2 methods could in fact be just 1 method

**Review by tmbp95 — APPROVED** [review]

**Pull request merged**